feat: agregar endpoints /mi-bodega para eventos y sucursales

NinoPassera opened the pull request:
- Crear GET /eventos/mi-bodega que filtra por bodega del usuario autenticado
- Crear GET /sucursales/mi-bodega que filtra por bodega del usuario autenticado
- Incluir relaciÃ³n bodega en respuestas de usuarios (GET /users/me y GET /users/:id)
- Agregar documentaciÃ³n OpenAPI completa para nuevos endpoints
- Usar req.bodegaId del token JWT para filtrado automÃ¡tico

NinoPassera committed:
feat: agregar endpoints /mi-bodega para eventos y sucursales

- Crear GET /eventos/mi-bodega que filtra por bodega del usuario autenticado
- Crear GET /sucursales/mi-bodega que filtra por bodega del usuario autenticado
- Incluir relaciÃ³n bodega en respuestas de usuarios (GET /users/me y GET /users/:id)

- Agregar documentaciÃ³n OpenAPI completa para nuevos endpoints
- Usar req.bodegaId del token JWT para filtrado automÃ¡tico

NinoPassera committed:
feat: mejorar nombres de permisos y agregar endpoint /rbac/roles/mi-bodega

- Renombrar permisos para mayor claridad:
  * SUDO â†’ ADMINISTRADOR_SISTEMA
  * BODEGAS_MANAGE â†’ ADMINISTRADOR_BODEGA
  * BODEGAS_VALIDATE â†’ SUPERVISOR_BODEGA
  * EVENTOS_MANAGE â†’ GESTOR_EVENTOS
  * RESERVAS_MANAGE â†’ GESTOR_RESERVAS

- Agregar endpoint /rbac/roles/mi-bodega que retorna:
  * Roles especÃ­ficos de la bodega del usuario
  * Roles globales del sistema (bodegaId = null)
  * Incluye middleware de autenticaciÃ³n y permisos

- Actualizar documentaciÃ³n OpenAPI
- Actualizar datos de seed con nuevos nombres
- Mejorar comentarios descriptivos en permisos
- Corregir tipos TypeScript para cumplir con ESLint

Tomas-Izuel commented on src/bodega/router.ts:
```diff
@@ -199,7 +199,7 @@ router.delete('/:id', controller.delete);
   '/:id/validate',
   authMiddleware,
-  requirePermissions([Permissions.BODEGAS_VALIDATE]),
+  requirePermissions([Permissions.SUPERVISOR_BODEGA]),
```
los permisos son por accion, no por puesto si no me equivoco

Emmanuel-Guerreiro commented:
No le cambies los nombres a los permisos del enum.

O sea si lo que querias era que en el frontend se vea con un nombre mas acorde, para eso la entidad de permiso tiene un name. Hay solo que udpatear el seed para que se vea bien

NinoPassera committed:
feat: agregar bodegaId al crear usuario

- Modificar la lógica en el método create del UsersController para incluir automáticamente el bodegaId del usuario autenticado en los datos del nuevo usuario.
- Asegurar que la creación de usuarios utilice la información del token JWT para una mejor gestión de datos.

NinoPassera committed:
refactor: actualizar sistema de permisos

- Actualizar enum Permissions con nueva nomenclatura consistente
- Cambiar GESTOR_EVENTOS por EVENTOS_MANAGE
- Cambiar GESTOR_RESERVAS por RESERVAS_MANAGE
- Cambiar SUPERVISOR_BODEGA por BODEGAS_VALIDATE
- Corregir referencias incorrectas a ADMINISTRADOR_SISTEMA por SUDO
- Actualizar documentaciÃ³n OpenAPI en routers
- Actualizar seed.ts con nuevos permisos
- Mantener compatibilidad total con funcionalidad existente

NinoPassera committed:
Resolve merge conflicts: mantiene funcionalidades de bodega y nuevas features de develop
